src/components/Subscriber/Subscriber.tsx

```diff
@@ -0,0 +1,100 @@
+import Input from '@/components/Input';
+import MultiSelect from '@/components/MultiSelect';
+import Button from '@/components/Button';
+import { getTags } from '@/apis/tag';
+import { useEffect, useState } from 'react';
+import { Formik, Form } from 'formik';
+import { subscriberValidationSchema } from '@/validations/subscriber';
+
+const Subscriber = ({
+  onSubmit,
+  selectedOptions,
+  setSelectedOptions,
+  subscriber = null,
+}: {
+  onSubmit: any;
+  selectedOptions: any;
+  setSelectedOptions: any;
+  subscriber: any;
+}) => {
+  const [tags, setTags] = useState([]);
+
+  useEffect(() => {
+    const fetchTagLists = async () => {
+      setTags(await getTags());
+    };
+
+    fetchTagLists();
+  }, []);
+
+  const tagsList =
+    Array.isArray(tags.data) && tags.data.length > 0
+      ? tags.data.map((tag: any) => ({
+          value: tag.id,
+          label: tag.name,
+        }))
+      : [];
+
+  const initialValues = {
+    email: subscriber?.email ? subscriber.email : '',
+    name: subscriber?.name ? subscriber.name : '',
+    phone: subscriber?.phone ? subscriber.phone : '',
+  };
+
+  function handleTagSelection(data: any) {
+    if (data.length > 0) {
+      setSelectedOptions(data);
+    } else {
+      setSelectedOptions([]);
+    }
+  }
+
+  return (
+    <>
+      <Formik
+        initialValues={initialValues}
+        onSubmit={onSubmit}
+        validationSchema={subscriberValidationSchema}
+        enableReinitialize={true}
```

Comment:
what does this mean?

Reply:
`enableReinitialize` is a formic props with the help of which we can reinitialize or update `initialValues`.